ui/components/data-sources/DataSourcesContainer.js

```diff
@@ -10,7 +10,8 @@ class DataSourcesContainer extends Component {
         super(props);
         this.state = {
             showModal: false,
-            modalText: null
+            modalText: null,
+            filter: {}
```

Comment:
Why an object and not just a string?

Reply:
The <Query> was failing with an error due to expecting an object and was getting a string. Changing it to an object solved the problem.

Reply:
The variables object could also have many variables so it expects an object containing these